spec/unit/application/master_spec.rb

```diff
@@ -314,11 +314,38 @@
         @master.main
       end
 
-      it "should drop privileges if running as root" do
-        Puppet.features.stubs(:root?).returns true
+      context "user privileges" do
+        before do
+          @user = Puppet::Type.type(:user).new(:name => 'puppet')
+          Puppet::Type.type(:user).expects(:new).returns @user
```

Comment:
I'm thinking we want to restrict this to only the cases where `.new` is called with `:name => 'puppet'`. We have some code that does something similar: https://github.com/puppetlabs/puppet/blob/master/spec/unit/settings_spec.rb#L1591-L1595. It also returns a mock `user` object, which I think is preferred when testing the master application.